geom/src/polygon.rs

```diff
@@ -348,23 +338,28 @@ impl Polygon {
         self.to_geo().intersects(&pl.to_geo())
     }
 
+    // TODO Temporary until we change the internal Polygon representation to always just be rings.
+    // Note this does expensive cloning right now
+    fn get_rings(&self) -> Vec<Ring> {
+        if let Some(ref rings) = self.rings {
+            rings.clone()
+        } else {
+            vec![Ring::must_new(self.points.clone())]
+        }
+    }
+
     /// Creates the outline around the polygon (both the exterior and holes), with the thickness
-    /// half straddling the polygon and half of it just outside. Only works for polygons that're
-    /// formed from rings.
+    /// half straddling the polygon and half of it just outside.
     ///
     /// Returns a `Tessellation` that may union together the outline from the exterior and multiple
     /// holes. Callers that need a `Polygon` must call `to_outline` on the individual `Rings`.
-    pub fn to_outline(&self, thickness: Distance) -> Result<Tessellation> {
-        if let Some(ref rings) = self.rings {
-            Ok(Tessellation::union_all(
-                rings
-                    .iter()
-                    .map(|r| Tessellation::from(r.to_outline(thickness)))
-                    .collect(),
-            ))
-        } else {
-            Ring::new(self.points.clone()).map(|r| Tessellation::from(r.to_outline(thickness)))
-        }
+    pub fn to_outline(&self, thickness: Distance) -> Tessellation {
```

Comment:
Most of the PR is just trivially fixing call-sites for this. We can always tessellate an outline for a polygon now!

Reply:
This is a nice cleanup! Though... I don't remember if all the things you are cleaning up were recently introduced as part of all this geometry-juggling, or if it was long standing.

Reply:
It's always been here, because a `Polygon` (like from SVG) might not've had rings that we can turn into outlines